src/clj/puppetlabs/services/jruby/jruby_puppet_service.clj

```diff
@@ -114,65 +99,28 @@
-
-(defmacro with-jruby-puppet
-  "Encapsulates the behavior of borrowing and returning an instance of
-  JRubyPuppet.  Example usage:
-
-  (let [jruby-service (get-service :JRubyPuppetService)]
-    (with-jruby-puppet
-      jruby-puppet
-      jruby-service
-      (do-something-with-a-jruby-puppet-instance jruby-puppet)))
-
-  Will throw an IllegalStateException if borrowing an instance of
-  JRubyPuppet times out."
-  [jruby-puppet jruby-service reason & body]
-  `(loop [pool-instance# (jruby/borrow-instance ~jruby-service ~reason)]
-     (if (nil? pool-instance#)
-       (sling/throw+
-         {:kind ::jruby-timeout
-          :msg (str "Attempt to borrow a JRuby instance from the pool "
-                    "timed out; Puppet Server is temporarily overloaded. If "
-                    "you get this error repeatedly, your server might be "
-                    "misconfigured or trying to serve too many agent nodes. "
-                    "Check Puppet Server settings: "
-                    "jruby-puppet.max-active-instances.")}))
-     (when (jruby-schemas/shutdown-poison-pill? pool-instance#)
-       (jruby/return-instance ~jruby-service pool-instance# ~reason)
-       (ringutils/throw-service-unavailable!
-         (str "Attempted to borrow a JRuby instance from the pool "
-              "during a shutdown. Please try again.")))
-     (if (jruby-schemas/retry-poison-pill? pool-instance#)
-       (do
-         (jruby/return-instance ~jruby-service pool-instance# ~reason)
-         (recur (jruby/borrow-instance ~jruby-service ~reason)))
-       (let [~jruby-puppet (:jruby-puppet pool-instance#)]
-         (try
-           ~@body
-           (finally
-             (jruby/return-instance ~jruby-service pool-instance# ~reason)))))))
-
-(defmacro with-lock
-  "Acquires a lock on the pool, executes the body, and releases the lock."
-  [jruby-service reason & body]
-  `(let [context# (-> ~jruby-service
-                      tk-services/service-context)
-         pool# (-> context#
-                   :pool-context
-                   core/get-pool)
-         event-callbacks# (:event-callbacks context#)]
-     (core/lock-pool pool# ~reason @event-callbacks#)
-     (try
-      ~@body
-      (finally
-        (core/unlock-pool pool# ~reason @event-callbacks#)))))
+    (let [pool-context (:pool-context (tk-services/service-context this))]
+      (jruby-core/register-event-handler pool-context callback-fn))))
+
+(def #^{:macro true
+        :doc "An alias for the jruby-utils' `with-jruby-instance` macro so
+             that it is accessible from the service namespace along with the
+             rest of the API."}
+  with-jruby-instance #'jruby-core/with-jruby-instance)
+
+(def #^{:macro true
+        :doc "An alias for the jruby-utils' `with-lock` macro so
+             that it is accessible from the service namespace along with the
+             rest of the API."}
+  with-lock #'jruby-core/with-lock)
```

Comment:
Is there any code left in this PR which exercises this?  If we think we might want to have it here just for the convenience of upstream consumers as opposed to just having them call into the  - e.g., PE Puppet Server code related to File Sync - might be good to at least have one test in OSS which confirms that this minimally does what it's supposed to do?